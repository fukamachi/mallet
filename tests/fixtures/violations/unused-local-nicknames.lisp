;; Test file for unused-local-nicknames rule

;; Case 1: One unused nickname among multiple
(defpackage #:test1/config
  (:use #:cl)
  (:local-nicknames
   (#:a #:alexandria)  ; Unused
   (#:glob #:trivial-glob))
  (:export #:load-config))
(in-package #:test1/config)

(defun load-config ()
  (glob:glob "*.conf"))

;; Case 2: Multiple unused nicknames
(defpackage #:test2/config
  (:use #:cl)
  (:local-nicknames
   (#:a #:alexandria)  ; Unused
   (#:glob #:trivial-glob)
   (#:unused #:some-package))  ; Unused
  (:export #:load-config))
(in-package #:test2/config)

(defun load-config ()
  (glob:glob "*.conf"))
