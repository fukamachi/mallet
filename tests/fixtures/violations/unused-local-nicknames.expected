# Expected violations for unused-local-nicknames.lisp
# Format: line:column rule-name severity

5:3 unused-local-nicknames info
23:3 unused-local-nicknames info
25:3 unused-local-nicknames info
